lib/deterministicmnlist/SimplifiedMNListStore.js

```diff
@@ -0,0 +1,149 @@
+ */
+function SimplifiedMNListStore(diffArray, options) {
+  const firstDiff = new SimplifiedMNListDiff(diffArray[0]);
+  this.network = firstDiff.network;
+  this.options = options;
+  this.maxDiffs = (typeof (options) !== 'undefined' && options.maxListsLimit) ? options.maxListsLimit : SMLSTORE_MAX_DIFFS;
+  this.diffStore = [];
+
+  // always store the first diff as a root to build on
+  this.baseSimplifiedMNList = new SimplifiedMNList(firstDiff);
+  this.currentSML = new SimplifiedMNList(firstDiff);
+  this.baseHeight = firstDiff.cbTx.extraPayload.height;
+  this.baseBlockHash = firstDiff.baseBlockHash;
+  this.updateTipInfo(this.baseHeight, firstDiff.blockHash);
+
+  if (!this.baseSimplifiedMNList) {
+    throw new Error('no initial diff has been provided');
+  }
+
+  // we only add to diffStore if more than one element is in diffArray
+  diffArray.forEach((diff, index) => {
+    if (index > 0) {
+      this.addDiff(diff);
+    }
+  });
+}
+
+/**
+ * Updates height and hash for latest block
+ * @param {number} height
+ * @param {string} hash
+ */
+SimplifiedMNListStore.prototype.updateTipInfo = function updateTipInfo(height, hash) {
+  this.tipHeight = height;
+  this.tipHash = hash;
+};
+
+/**
+ * Adds a new SimplifiedMNListDiff to the store.
+ * If SMLSTORE_MAX_DIFFS reached it drops the oldest
+ * @param {SimplifiedMNListDiff} diff
+ */
+SimplifiedMNListStore.prototype.addDiff = function addDiff(diff) {
+  if (!this.baseSimplifiedMNList) {
+    throw new Error('no initial diff has been provided');
+  }
+
+  if (this.diffStore.length >= this.maxDiffs - 1) {
+    this.baseSimplifiedMNList.applyDiff(this.diffStore[0].diff);
+    this.diffStore.shift();
+  }
+
+  const cbTx = new Transaction(diff.cbTx);
+  const { height } = cbTx.extraPayload;
+  this.diffStore.push({ height, diff });
+  this.currentSML.applyDiff(diff);
+  this.updateTipInfo(height, diff.blockHash);
+};
+
+/**
+ * Returns height for latest block
+ * @returns {number}
+ */
+SimplifiedMNListStore.prototype.getTipHeight = function getTipHeight() {
+  return this.tipHeight;
+};
+
+/**
+ * Returns hash for latest block
+ * @returns {string}
+ */
+SimplifiedMNListStore.prototype.getTipHash = function getTipHash() {
+  return this.tipHash;
+};
+
+/**
+ * Returns a SimplifiedMNListDiff entry by height
+ * @param {number} startHeight
+ * @param {number} endHeight
+ * @returns {SimplifiedMNListDiff[]}
+ */
+SimplifiedMNListStore.prototype.getSMLDiffbyHeightRange = function getSMLDiffbyHeightRange(startHeight, endHeight) {
+  return this.diffStore.filter(
+    diff => diff.height >= startHeight && diff.height <= endHeight,
+  ).map(
+    filteredDiffs => filteredDiffs.diff,
+  );
+};
+
+/**
+ * Returns a SimplifiedMNList by block height
+ * @param {number} height
+ * @returns {SimplifiedMNList}
+ */
+SimplifiedMNListStore.prototype.getSMLbyHeight = function getSMLbyHeight(height) {
+  const diffs = (this.baseHeight === height)
```

Comment:
What if base list on this height?

Reply:
then we just return the base SML. Do you want me to refactor without the ternary expr. and return early with `this.baseSimplifiedMNList`?

Reply:
Oh, no. I just didn't get the logic. Everything is fine then.